internal/spot/spotinst_services_ocean_spark.go

```diff
@@ -0,0 +1,127 @@
+package spot
+
+import (
+	"context"
+	"strings"
+
+	"github.com/spotinst/spotinst-sdk-go/service/ocean/spark"
+	"github.com/spotinst/spotinst-sdk-go/spotinst"
+
+	"github.com/spotinst/spotctl/internal/log"
+)
+
+type oceanSpark struct {
+	svc spark.Service
+}
+
+func (x *oceanSpark) ListClusters(ctx context.Context, controllerClusterID string, state string) ([]*OceanSparkCluster, error) {
+	log.Debugf("Listing Ocean Spark clusters")
+
+	filter := &spark.ListClustersInput{}
+
+	if controllerClusterID != "" {
+		filter.ControllerClusterID = spotinst.String(controllerClusterID)
+	}
+
+	if state != "" {
+		filter.ClusterState = spotinst.String(strings.ToUpper(state))
+	}
+
+	output, err := x.svc.ListClusters(ctx, filter)
+	if err != nil {
+		return nil, err
+	}
+
+	clusters := make([]*OceanSparkCluster, len(output.Clusters))
+	for i, cluster := range output.Clusters {
+		clusters[i] = &OceanSparkCluster{
+			TypeMeta: TypeMeta{
+				Kind: typeOf(OceanSparkCluster{}),
+			},
+			ObjectMeta: ObjectMeta{
+				ID:        spotinst.StringValue(cluster.ID),
+				Name:      spotinst.StringValue(cluster.ControllerClusterID),
+				CreatedAt: spotinst.TimeValue(cluster.CreatedAt),
+				UpdatedAt: spotinst.TimeValue(cluster.UpdatedAt),
+			},
+			OceanClusterID: spotinst.StringValue(cluster.OceanClusterID),
+			State:          spotinst.StringValue(cluster.State),
+			Obj:            cluster,
+		}
+	}
+
+	return clusters, nil
+}
+
+func (x *oceanSpark) GetCluster(ctx context.Context, clusterID string) (*OceanSparkCluster, error) {
+	log.Debugf("Getting Ocean Spark cluster by ID: %s", clusterID)
+
+	input := &spark.ReadClusterInput{
+		ClusterID: spotinst.String(clusterID),
+	}
+
+	output, err := x.svc.ReadCluster(ctx, input)
+	if err != nil {
+		return nil, err
+	}
+
+	cluster := &OceanSparkCluster{
+		TypeMeta: TypeMeta{
+			Kind: typeOf(OceanSparkCluster{}),
+		},
+		ObjectMeta: ObjectMeta{
+			ID:        spotinst.StringValue(output.Cluster.ID),
+			Name:      spotinst.StringValue(output.Cluster.ControllerClusterID),
+			CreatedAt: spotinst.TimeValue(output.Cluster.CreatedAt),
+			UpdatedAt: spotinst.TimeValue(output.Cluster.UpdatedAt),
+		},
+		Obj: output.Cluster,
```

Comment:
Wouldn't we want the `OceanClusterID` and `State` set for single retrievals as well ?

Reply:
The single retrieval is only used in the describe operation, which prints out the whole json.
This is consistent with other spotctl get/describe operations.